Classes/NSManagedObject+DataMapping.m

```diff
@@ -373,9 +373,9 @@ + (NSDictionary*)processCollectionOfEntitiesAccordingToStatusIndicator:(NSArray*
 + (void) parseAttributes:(NSString*)attributeName mappedItemAttributes:(NSArray*)mappedItemAttributes objAttributes:(NSDictionary*)objAttributes
                      obj:(NSManagedObject*)obj convertedAttributes:(NSMutableDictionary*)convertedAttributes
 {
+    // Only process the key if it's in the remote dictionary, or if it's null (to persist null to the DB).
     NSString *remoteKey = [mappedItemAttributes objectAtIndex:kKernArrayIndexRemoteKey];
-    // only process key if it's in our provided set
-    if ([[objAttributes allKeys] containsObject:remoteKey]) {
+    if ([objAttributes isEqual:[NSNull null]] || ([objAttributes isKindOfClass:[NSDictionary class]] && [[objAttributes allKeys] containsObject:remoteKey])) {
```

Comment:
Why would we not persist the null to the DB as the test shows? If that check is removed, then `testUpdatesAnExistingEntityWithRemoteDictionaryNestedNull` will fail because `firstName` and `lastName` will still be `That` and `Guy`.